feature/bgp/otg_tests/bgp_override_as_path_split_horizon_test/bgp_override_as_path_split_horizon_test.go

```diff
@@ -0,0 +1,581 @@
+		}
+	}
+}
+
+// bgpCreateNbr creates a BGP object with neighbors pointing to atePort1 and atePort2
+func bgpCreateNbr(t *testing.T, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	t.Helper()
+	dutOcRoot := &oc.Root{}
+	ni1 := dutOcRoot.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	niProto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := niProto.GetOrCreateBgp()
+	global := bgp.GetOrCreateGlobal()
+	global.RouterId = ygot.String(dutPort2.IPv4)
+	global.As = ygot.Uint32(dutGlobalAS)
+
+	// Note: we have to define the peer group even if we aren't setting any policy because it's
+	// invalid OC for the neighbor to be part of a peer group that doesn't exist.
+
+	for _, nbr := range []*bgpNeighbor{nbr1, nbr2} {
+
+		pg := bgp.GetOrCreatePeerGroup(nbr.peerGrp)
+		pg.PeerAs = ygot.Uint32(nbr.peerAS)
+		pg.LocalAs = ygot.Uint32(nbr.localAS)
+		pg.PeerGroupName = ygot.String(nbr.peerGrp)
+
+		nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
+		nv4.PeerGroup = ygot.String(nbr.peerGrp)
+		nv4.PeerAs = ygot.Uint32(nbr.peerAS)
+		nv4.Enabled = ygot.Bool(true)
+		af4 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+		af4.Enabled = ygot.Bool(true)
+
+		if deviations.RoutePolicyUnderAFIUnsupported(dut) {
+			rpl := pg.GetOrCreateApplyPolicy()
+			rpl.ImportPolicy = []string{policyName}
+			rpl.ExportPolicy = []string{policyName}
+		} else {
+			pgaf := pg.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+			pgaf.Enabled = ygot.Bool(true)
+			rpl := pgaf.GetOrCreateApplyPolicy()
+			rpl.ImportPolicy = []string{policyName}
+			rpl.ExportPolicy = []string{policyName}
+		}
+	}
+	return niProto
+}
+
+// verifyBgpTelemetry checks that the dut has an established BGP session with reasonable settings.
+func verifyBgpTelemetry(t *testing.T, dut *ondatra.DUTDevice, nbrsList []*bgpNeighbor) {
+	t.Helper()
+	t.Logf("Verifying BGP state.")
+	bgpPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+
+	for _, nbr := range nbrsList {
+		nbrPath := bgpPath.Neighbor(nbr.neighborip)
+		t.Logf("Waiting for BGP neighbor to establish...")
+		status, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+			state, ok := val.Val()
+			return ok && state == oc.Bgp_Neighbor_SessionState_ESTABLISHED
+		}).Await(t)
+		if !ok {
+			t.Fatal("No BGP neighbor formed")
+		}
+		state, _ := status.Val()
+		t.Logf("BGP adjacency for %s: %v", nbr.neighborip, state)
+		if want := oc.Bgp_Neighbor_SessionState_ESTABLISHED; state != want {
+			t.Errorf("BGP peer %s status got %d, want %d", nbr.neighborip, state, want)
+		}
+	}
+}
+
+// configureOTG configures the interfaces and BGP protocols on an ATE.
+func configureOTG(t *testing.T, otg *otg.OTG) (gosnappi.BgpV4Peer, gosnappi.DeviceIpv4, gosnappi.Config) {
```

Comment:
can be obsoleted by NewBGPSession

Reply:
I am using specific AS numbers for this test and these tests have predefined AS numbers , hence using local functions.

Reply:
Ok.